core/src/main/scala/com/salesforce/op/filters/RawFeatureFilterResults.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2017, Salesforce.com, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.filters
+
+import com.salesforce.op.stages.impl.preparators.CorrelationType
+import com.salesforce.op.utils.json.EnumEntrySerializer
+import org.json4s.jackson.Serialization
+import org.json4s.{DefaultFormats, Formats}
+
+import scala.util.Try
+
+/**
+ * Contains configuration and results from RawFeatureFilter
+ *
+ * @param rawFeatureFilterConfig  configuration settings for RawFeatureFilter
+ * @param rawFeatureDistributions feature distributions calculated from training data
+ * @param rawFeatureFilterMetrics feature metrics calculated by RawFeatureFilter
+ * @param exclusionReasons        results of RawFeatureFilter tests (reasons why feature is dropped or not)
+ */
+case class RawFeatureFilterResults
+(
+  rawFeatureFilterConfig: RawFeatureFilterConfig = RawFeatureFilterConfig(),
+  rawFeatureDistributions: Seq[FeatureDistribution] = Seq.empty,
+  rawFeatureFilterMetrics: Seq[RawFeatureFilterMetrics] = Seq.empty,
+  exclusionReasons: Seq[ExclusionReasons] = Seq.empty
+) extends RawFeatureFilterResultsLike
+
+object RawFeatureFilterResults {
+
+  implicit val jsonFormats: Formats = DefaultFormats +
```

Comment:
put it there i think

Reply:
@leahmcguire @Jauntbox 

`OpPipelineStageReadWriteShared` is in the `features` module, whereas `RawFeatureFilter` files are all in `core`. I think if we decide to move the `Formats` variable to `OpPipelineStageReadWriteShared`, then we may have to do some refactoring to avoid some circular reference? Like move `RawFeatureFilter` and all associated files into the `features` module

Please let me know if you know of a way to do this without having to refactor, or how else you'd like to proceed (keep as is, move the `Formats` variable to their respective classes, etc.)

For reference, this is what is in `OpPipelineStageReadWriteShared`:

```
implicit val formats: Formats =
    DefaultFormats ++
      JodaTimeSerializers.all +
      EnumEntrySerializer.json4s[AnyValueTypes](AnyValueTypes) +
      EnumEntrySerializer.json4s[HashAlgorithm](HashAlgorithm) +
      EnumEntrySerializer.json4s[HashSpaceStrategy](HashSpaceStrategy) +
      EnumEntrySerializer.json4s[ScalingType](ScalingType) +
      EnumEntrySerializer.json4s[TimePeriod](TimePeriod) +
      EnumEntrySerializer.json4s[FeatureDistributionType](FeatureDistributionType) +
      new SpecialDoubleSerializer
```

Reply:
good point, but it turns out that we actually have all the serialization formats for model insights in the companion object so lets move it there for consistency.

https://github.com/salesforce/TransmogrifAI/blob/master/core/src/main/scala/com/salesforce/op/ModelInsights.scala#L391

Reply:
@leahmcguire Thanks for pointing this example out!

Serialization formats for `RawFeatureFilterResults` is currently placed in a companion object. So just wondering what you're envisioning if beyond what's implemented already

Couple possibilities below:

- There are some `Formats` lines of `OpPipelineStageReadWriteShared` that correspond to an existing class (e.g., `ScalingType`). We can move these into companion objects

- SerializationFormats in `RawFeatureFilterResults` currently extends from `DefaultFormats` but the one in `ModelInsights` does not. We can make `RawFeatureFilterResults` better resemble `ModelInsights` by explicitly defining what it needs from  `DefaultFormats`?

Thanks!

Reply:
@leahmcguire @clin-projects Is `RawFeatureFilterResults` going to be serialized as a part of `ModelInsights`?

Reply:
Yes it is. Currently raw feature distributions and exclusion reasons are serialized as part of `FeatureInsights`. Next step is serializing raw feature metrics and RFF configuration. Configuration will be incorporated in `getStageInfo`